### PR TITLE
Object Store / Key Value coverage

### PR DESCRIPTION
**Pull request by scottf**
Adding Object Store coverage found some minor issues so those were fixed.
Key Value added some coverage.

### CODE REVIEW

File: src/main/java/io/nats/client/api/ObjectInfo.java [hunk 1]
```diff
@@ -181,6 +181,7 @@ public Builder(ObjectInfo info) {
             modified = info.modified;
             chunks = info.chunks;
             digest = info.digest;
+            deleted = info.deleted;
             metaBuilder = ObjectMeta.builder(info.objectMeta);
```
Comment (scottf) [inline src/main/java/io/nats/client/api/ObjectInfo.java:185]: Unit Testing found this!

File: src/main/java/io/nats/client/impl/NatsKeyValue.java [hunk 2]
```diff
@@ -48,13 +48,11 @@ public class NatsKeyValue extends NatsFeatureBase implements KeyValue {
...
+            pubSubKeyPrefix = rawKeyPrefix;
+        }
         else {
-            if (kvo.getJetStreamOptions().isDefaultPrefix()) {
-                pubSubKeyPrefix = rawKeyPrefix;
-            }
-            else {
-                pubSubKeyPrefix = kvo.getJetStreamOptions().getPrefix() + rawKeyPrefix;
-            }
+            pubSubKeyPrefix = kvo.getJetStreamOptions().getPrefix() + rawKeyPrefix;
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsKeyValue.java:55]: Simplifiied.

File: src/main/java/io/nats/client/impl/NatsObjectStore.java [hunk 3]
```diff
@@ -50,15 +50,13 @@ public class NatsObjectStore extends NatsFeatureBase implements ObjectStore {
...
-            if (oso.getJetStreamOptions().isDefaultPrefix()) {
-                pubSubChunkPrefix = rawChunkPrefix;
-                pubSubMetaPrefix = rawMetaPrefix;
-            }
-            else {
-                pubSubChunkPrefix = oso.getJetStreamOptions().getPrefix() + rawChunkPrefix;
-                pubSubMetaPrefix = oso.getJetStreamOptions().getPrefix() + rawMetaPrefix;
-            }
+            pubSubChunkPrefix = oso.getJetStreamOptions().getPrefix() + rawChunkPrefix;
+            pubSubMetaPrefix = oso.getJetStreamOptions().getPrefix() + rawMetaPrefix;
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsObjectStore.java:59]: Simplified

File: src/main/java/io/nats/client/impl/NatsObjectStore.java [hunk 4]
```diff
@@ -159,6 +157,9 @@ public ObjectInfo put(ObjectMeta meta, InputStream inputStream) throws IOExcepti
 
             throw e;
         }
+        finally {
+            try { inputStream.close(); } catch (IOException ignore) {}
+        }
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsObjectStore.java:162]: Be a good citizen. Not sure what to do if the close fails, it's not really relevant at that point anyway.

File: src/test/java/io/nats/client/impl/KeyValueTests.java [hunk 5]
```diff
@@ -49,6 +49,7 @@ public void testWorkflow() throws Exception {
         runInJsServer(nc -> {
             // get the kv management context
             KeyValueManagement kvm = nc.keyValueManagement();
+            nc.keyValueManagement(KeyValueOptions.builder(DEFAULT_JS_OPTIONS).build()); // coverage
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/KeyValueTests.java:52]: I must have coverage!

**Review by RichardHightower — APPROVED** [review]
LGTM

**Pull request merged**